extensions/json-format/src/formatSelectionAndShow.tsx

```diff
@@ -0,0 +1,20 @@
+import { getSelectedText, useNavigation } from "@raycast/api";
+import { formatJS } from "./utils";
+import { FormattedJsonDetail } from "./formattedJsonDetail";
+import { useEffect } from "react";
+
+export default function Command() {
+  const { push } = useNavigation();
+
+  useEffect(() => {
+    async function format() {
+      const output = await formatJS(await getSelectedText());
+      if (output) {
+        push(<FormattedJsonDetail json={output} />);
+      }
+    }
+    format();
+  }, [push]);
+
+  return null;
```

Comment:
Instead of returning `null` and using `push`, can't we return _FormattedJsonDetail_ directly? Something like this:

```ts
export default function Command() {
  const {data: output} = usePromise(async () => await formatJS(await getSelectedText()));
  return <FormattedJsonDetail json={output || []} />;
}
```

Reply:
Thank you for your suggestion.